lib/ansible/modules/packaging/language/pip.py

```diff
@@ -235,6 +241,73 @@
                              'pip': 'import pkg_resources; print(pkg_resources.get_distribution("pip").version)'}
 
 
+class Distribution:
+    def __init__(self, name_string, version_string=None):
+        if not name_string.startswith(('svn+', 'git+', 'hg+', 'bzr+', 'file:')):
+            self._plain_distribution = True
+            if version_string:
+                version_string = version_string.lstrip()
+                separator = '==' if version_string[0].isdigit() else ' '
+                name_string = separator.join((name_string, version_string))
+            self._requirement = Requirement.parse(name_string)
+            self._distribution_name = self._requirement.project_name
+        else:
+            self._plain_distribution = False
+            self._distribution_name = name_string
+
+    @property
+    def has_version_specifier(self):
+        if self._plain_distribution:
+            return bool(self._requirement.specs)
+        return False
+
+    @property
+    def distribution_name(self):
+        return self._distribution_name
+
+    @distribution_name.setter
+    def distribution_name(self, new_name):
+        self._distribution_name = new_name
+
+    def is_satisfied_by(self, version_to_test, module):
```

Comment:
Look what I've found:

```python
$ ipython
Python 3.7.0 (default, Jun 28 2018, 11:18:11)
Type 'copyright', 'credits' or 'license' for more information
IPython 6.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import pkg_resources

In [10]: ipdst = pkg_resources.get_distribution('ipython')

In [11]: ipdst
Out[11]: ipython 6.4.0 (/home/wk/.pyenv/versions/3.7.0/envs/ansible-py3.7.0-pyenv-venv/lib/python3.7/site-packages)

In [15]: ipdst.as_requirement()
Out[15]: Requirement.parse('ipython==6.4.0')

```

This probably can completely eliminate need for doing `pip freeze` and parsing it!

@HD650 please investigate this.

Reply:
```python
$ python2.6
Python 2.6.9 (unknown, Apr 10 2018, 17:32:50)
[GCC 7.3.0] on linux4
Type "help", "copyright", "credits" or "license" for more information.
>>> import pkg_resources
/home/wk/.pyenv/versions/2.6.9/lib/python2.6/site-packages/pkg_resources.py:17: DeprecationWarning: the sets module is deprecated
  from sets import ImmutableSet
>>> stdst = pkg_resources.get_distribution('setuptools')
>>> stdst
setuptools 0.6b4dev-r0 (/home/wk/.pyenv/versions/2.6.9/lib/python2.6/site-packages)
>>> stdst.as_requirement()
Requirement.parse('setuptools==0.6b4dev-r0')
>>> stdst.as_requirement().specs
[('==', '0.6b4dev-r0')]
```